pkg/controller/account/account_controller.go

```diff
@@ -643,9 +643,23 @@ func (r *ReconcileAccount) assumeRole(
 	roleToAssume string,
 	ccsRoleID string) (awsclient.Client, *sts.AssumeRoleOutput, error) {
 
+	var roleArn string
+
+	// get configMap and check if account is fedramp
+	configMap, err := controllerutils.GetOperatorConfigMap(r.Client)
+	if err != nil {
+		log.Error(err, "failed retrieving configmap")
+	}
+	_ , ok := configMap.Data["fedramp"]
+	if ok {
+		fmt.Sprintln("Account is govcloud")
+		roleArn = fmt.Sprintf("arn:aws-us-gov:iam::%s:role/%s", currentAcctInstance.Spec.AwsAccountID, roleToAssume)
+		// log.Error(err, "fedramp key not available in configmap")
+	}
+
 	// The role ARN made up of the account number and the role which is the default role name
 	// created in child accounts
-	var roleArn = fmt.Sprintf("arn:aws:iam::%s:role/%s", currentAcctInstance.Spec.AwsAccountID, roleToAssume)
+	roleArn = fmt.Sprintf("arn:aws:iam::%s:role/%s", currentAcctInstance.Spec.AwsAccountID, roleToAssume)
```

Comment:
This would need to be in an "else" block otherwise it will overwrite the fedramp role 100% of the time.

Reply:
agreed. changed by setting this beforehand